cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/JenaModelSparqlService.java

```diff
@@ -58,7 +59,7 @@ public void upload(Model model, String graphUri) {
 
     private Dataset getDatasetForSelect() {
         if (simulateRelaxedVirtuosoSparqlSelect) {
-            Dataset relaxedDataset = DatasetFactory.create(dataset);
+            Dataset relaxedDataset = DatasetFactory.wrap(DatasetGraphFactory.cloneStructure(dataset.asDatasetGraph()));
```

Comment:
why do we need to clone the DatasetGraph?

Reply:
Well, I only rewrote the original logic ... Personally, I don't see any reason for cloning it either, but maybe @natan-cox-cognizone [knows who knows](https://github.com/cognizone/asquare/blame/554c0a697cf172f62fa194dc1dd0ec1da8a20b8e/cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/JenaModelSparqlService.java#L61) about some :-D ...

```
 public static Dataset create(Dataset dataset) {
    Objects.requireNonNull(dataset, "Clone dataset is null");
    return new DatasetImpl(dataset);
  }
```
...
```
public DatasetImpl(Dataset ds) {
    this(DatasetGraphFactory.cloneStructure(ds.asDatasetGraph()));
}
  ```

Reply:
Ah. Well from the changes it did look like you purposely added the wrapper. If that is not the case then I would indeed keep it as it is. Seems, like there was a reason.